Assets/MixedRealityToolkit/UX/Scripts/BoundingBoxes/BoundingBoxRig.cs

```diff
@@ -0,0 +1,470 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using MixedRealityToolkit.InputModule.Utilities.Interations;
+using MixedRealityToolkit.UX.BoundingBoxes;
+using MixedRealityToolkit.Examples.InputModule;
+using MixedRealityToolkit.UX.Buttons;
+using MixedRealityToolkit.UX.AppBarControl;
+using UnityEngine;
+
+namespace MixedRealityToolkit.UX.BoundingBoxes
+{
+    public class BoundingBoxRig : MonoBehaviour
+    {
+        [Header("Flattening")]
+        [SerializeField]
+        [Tooltip("Choose this option if Rig is to be applied to a 2D object.")]
+        private BoundingBox.FlattenModeEnum flattenedAxis;
+
+        [Header("Customization Settings")]
+        [SerializeField]
+        private Material scaleHandleMaterial;
+        [SerializeField]
+        private Material rotateHandleMaterial;
+        [SerializeField]
+        private Material interactingMaterial;
+
+        [Header("Preset Components")]
+        [SerializeField]
+        [Tooltip("To visualize the object bounding box, drop the MixedRealityToolkit/UX/Prefabs/BoundingBoxes/BoundingBoxBasic.prefab here.")]
+        public BoundingBox BoundingBoxPrefab;
```

Comment:
if this is serialized, then it should also be private.
if access outside the class is needed, then please create a property and use the field as it's backing.